nautobot/extras/context_managers.py

```diff
@@ -116,13 +121,34 @@ def web_request_context(user, context_detail="", change_id=None):
     :param user: User object
     :param context_detail: Optional extra details about the transaction (ex: the plugin name that initiated the change)
     :param change_id: Optional uuid object to uniquely identify the transaction. One will be generated if not supplied
+    :param context: Optional string value of the generated change log entries' "change_context" field, defaults to ObjectChangeEventContextChoices.CONTEXT_ORM.
+        Valid choices are in nautobot.extras.choices.ObjectChangeEventContextChoices
+    :param request: Optional web request instance, one will be generated if not supplied
     """
+    from nautobot.extras.jobs import enqueue_job_hooks  # prevent circular import
 
-    if not isinstance(user, get_user_model()):
-        raise TypeError("The user object must be an instance of nautobot.users.models.User")
+    valid_contexts = {
+        ObjectChangeEventContextChoices.CONTEXT_JOB: JobChangeContext,
+        ObjectChangeEventContextChoices.CONTEXT_JOB_HOOK: JobHookChangeContext,
+        ObjectChangeEventContextChoices.CONTEXT_ORM: ORMChangeContext,
+        ObjectChangeEventContextChoices.CONTEXT_WEB: WebChangeContext,
+    }
 
-    request = RequestFactory().request(SERVER_NAME="web_request_context")
-    request.user = user
-    change_context = ORMChangeContext(request=request, context_detail=context_detail, change_id=change_id)
-    with change_logging(change_context):
-        yield request
+    if context not in valid_contexts:
+        raise TypeError(f"{context} is not a valid context")
+
+    if not isinstance(user, (get_user_model(), AnonymousUser)):
+        raise TypeError(f"{user} is not a valid user object")
+
+    if request is None:
+        request = RequestFactory().request(SERVER_NAME="web_request_context")
+        request.user = user
+    change_context = valid_contexts[context](request=request, context_detail=context_detail, change_id=change_id)
+    try:
+        with change_logging(change_context):
+            yield request
+    finally:
+        # enqueue jobhooks and webhooks
+        for object_change in ObjectChange.objects.filter(request_id=change_context.change_id).iterator():
```

Comment:
Is `change_context.change_id` different from `change_id`, or should we use the latter for simplicity?

Reply:
It could be. If `change_id` is `None` the change_context generates one.

Reply:
Thanks for the info! Maybe add a comment to this effect?